File: Leetcode/14. Longest Common Prefix.ts
function longestCommonPrefix(strs: string[]): string {
    let ans: string = strs[0];
    
    for(let i=1; i<strs.length; i++) {
        let temp: string = '';
        
        for(let j=0; j<Math.min(ans.length, strs[i].length); j++) {
            if(ans[j] !== strs[i][j]) break;
            temp += ans[j];
        }
        
        ans = temp;
    }
    
    return ans;
};

File: Leetcode/C++/14. Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans = strs[0];
        for(auto str : strs) {
            int j=0;
            string t;
            while(j<str.size() && j<ans.size()) {
                if(str[j] != ans[j]) break;
                t += ans[j++];
            }
            ans = t;
        }
        return ans;
    }
};
